framework/math/VectorGhostCommunicator/vector_ghost_communicator.h

```diff
@@ -28,35 +34,40 @@ class VectorGhostCommunicator
 
   MPI_Comm Communicator() const { return comm_; }
 
-  int64_t MapGhostToLocal(const int64_t ghost_id) const;
+  int64_t MapGhostToLocal(int64_t ghost_id) const;
 
   void CommunicateGhostEntries(std::vector<double>& ghosted_vector) const;
 
   std::vector<double> MakeGhostedVector() const;
   std::vector<double>
   MakeGhostedVector(const std::vector<double>& local_vector) const;
 
-private:
-  int FindOwnerPID(const int64_t global_id) const;
-
 protected:
   const uint64_t local_size_;
   const uint64_t global_size_;
   const std::vector<int64_t> ghost_ids_;
   const MPI_Comm comm_;
 
-  int location_id_;
-  int process_count_;
-  std::vector<uint64_t> extents_;
+  const int location_id_;
+  const int process_count_;
+  const std::vector<uint64_t> extents_;
+
+  struct CachedParallelData
+  {
+    std::vector<int> sendcounts_;
+    std::vector<int> senddispls_;
+    std::vector<int> recvcounts_;
+    std::vector<int> recvdispls_;
 
-  std::vector<int> sendcounts_;
-  std::vector<int> senddispls_;
-  std::vector<int> recvcounts_;
-  std::vector<int> recvdispls_;
+    std::vector<int64_t> local_ids_to_send_;
+    std::map<int64_t, size_t> ghost_to_recv_map_;
+  };
 
-  std::vector<int64_t> local_ids_to_send_;
-  std::map<int64_t, size_t> ghost_to_recv_map_;
+  const CachedParallelData cached_parallel_data_;
```

Comment:
Consider declaring the struct at the top of the class.

Reply:
Nothing is needed, just formatting. I always like to do 
```
class <class_name>
{
public/protected/private:
  typedefs ...
public/protected/private:
  structs ...
public/protected/private:
  functions ...
public/protected/private:
  members ...
};
```
instead of organizing by public/protected/private.